com.oracle.truffle.r.runtime/src/com/oracle/truffle/r/runtime/ffi/FFIMaterializeNode.java

```diff
@@ -131,9 +133,15 @@ protected static Object wrap(RBaseObject value, @SuppressWarnings("unused") bool
     }
 
     @Specialization
-    protected static Object wrap(RIntVector value, @SuppressWarnings("unused") boolean protect) {
+    protected static Object wrap(RIntVector value, @SuppressWarnings("unused") boolean protect,
+                                 @Cached("createBinaryProfile()") ConditionProfile isAltrepProfile) {
         // TODO specialize only for sequences (and maybe some other)
-        return value.cachedMaterialize();
+        if (isAltrepProfile.profile(value.isAltRep())) {
```

Comment:
Tohle jsem sem pridal, protoze to pro alt integer vektory zpusobovalo nekonecnou rekurzi a tudiz StackOverflow